docs/TuxReact.md

```diff
@@ -0,0 +1,449 @@
+
+In the following examples we will be writing in basic `React` syntax.
+
+***
+
+#### <a id="Rendering-in-React"></a>2) Rendering in React [#](#Rendering-in-React)
+The most common way to render a component in `React` is with the `React.render` method:
+
+```
+    var React = require('react');
+    var MyComponent = require('./components/MyComponent'); //Some component we want to render
+    React.render(<MyComponent />, document.body); //Using JSX for MyComponent
+```
+
+Oftentimes, if you are using the `tux/Router`, you will only need to invoke this method once to render the top level `Tux/Router/run` onto the DOM. This component will define your client-side application routes which will automatically render their child components when the corresponding route is reached:
+
+```
+    var React = require('tux/React');
+    var routes = require('./routes'); //your application routes
+    var run = require('tux/Router/run');
+
+    run(routes, function (Handler) {
+      React.render(<Handler />, document.body); //Again using JSX for Handler
+    });
+```
+
+At that point everything else will be rendered into the DOM through the `render` methods within your individual components. Let's dive into these `React` components.
+
+***
+
+#### <a id="React-Components"></a>3) React Components [#](#React-Components)
+`React` components are defined via the `React.createClass` method. Pass in an object to this method, and `React` will return a class which you can then use to render components built from that class into the DOM:
+
+```
+    var React = require('react');
+    var MyComponent = React.createClass({
+      render: function () {
+        return (
+          <h1>Hello World!</h1>
+        );
+      }
+    });
+```
+
+***
+
+#### <a id="React-Components-the-Render-Method"></a>3.1) the Render Method [#](#React-Components-the-Render-Method)
+All `React` classes must have a `render` method. The `render` method must return a single `React` component (either a basic `React` html component for rendering onto the DOM (`h1`, `div`, `p`, etc) or a `React` composite class (any custom build `React` component, for example `MyComponent` defined above is a `React` composite class).
+
+It might seem like returning only a single `React` component is a bit limiting. However, you can pass in child components to whatever `React` component you render. Will get more into what that actually means in the next section, but in laymens terms it means you can `render` however many components you want:
+
+```
+    var React = require('react');
+    var MyComplicatedComponent = React.createClass({
+      var someArrayOfComponents = [<ComponentA key={0} />, <ComponentB key={1} />];
+      render: function () {
+        return (
+          <div>
+            <MyComponent className='custom-class-name' />
+            <h1>Let's render something {thisComponentIsAwesome ? 'Awesome' : 'Lame'}</h1>
```

Comment:
thisComponentIsAwesome isn't defined anywhere in the example, so I'd recommend adding in a declaration at the beginning of the render and setting it to a boolean.